google-cloud-bigquery/acceptance/bigquery/dataset_access_test.rb

```diff
@@ -161,4 +162,292 @@
       refute dataset.access.reader_routine? routine
     end
   end
+
+  describe :condition do
+    let(:dataset_id) { "#{SecureRandom.hex(4)}_dataset" }
+    let(:dataset) do
+      d = bigquery.dataset dataset_id, access_policy_version: 3
+      if d.nil?
+        d = bigquery.create_dataset dataset_id, access_policy_version: 3
+      end
+      d
```

Comment:
Nit: a common Ruby idiom is the use of the `||` operator to set a value from several options that could either return a value or nil, similar to what you might do in a shell script. In this case:

```ruby
let(:dataset) do
  bigquery.dataset(dataset_id, access_policy_version: 3) ||
    bigquery.create_dataset(dataset_id, access_policy_version: 3)
end
```

(You need to include the parentheses here because of precedence rules between the method calls and the `||` operator. Rubocop knows this and will not flag the parens here.)